common/ManagementTestShared/MgmtRecordTestBase.cs

```diff
@@ -0,0 +1,224 @@
+
+        [OneTimeTearDown]
+        public async Task RunOneTimeTearDown()
+        {
+            StopTestRecording();
+            await CleanupResourceGroupsAsync();
+            Logger?.Dispose();
+            Logger = null;
+            await OnOneTimeTearDownScenarioAsync();
+            await OnOneTimeTearDownBaseAsync();
+        }
+
+        [SetUp]
+        public void StartTestRecording()
+        {
+            StopTestRecording();
+            TestContext.TestAdapter test = TestContext.CurrentContext.Test;
+            if (Mode != RecordedTestMode.Live &&
+                test.Properties.ContainsKey("SkipRecordings"))
+            {
+                throw new IgnoreException((string)test.Properties.Get("SkipRecordings"));
+            }
+            Recording = new TestRecording(Mode, GetSessionFilePath(), Sanitizer, Matcher);
+            TestEnvironment.Mode = Mode;
+            TestEnvironment.SetRecording(Recording);
+            InitializeClients();
+            OnSetupBaseAsync();
+            OnOneTimeTearDownScenarioAsync();
+        }
+
+        [TearDown]
+        public void StopTestRecording()
+        {
+            bool save = TestContext.CurrentContext.Result.FailCount == 0;
+#if DEBUG
+            save |= SaveDebugRecordingsOnFailure;
+#endif
+            Recording?.Dispose(save);
+            OnTearDownScenarioAsync();
+            OnTearDownScenarioAsync();
+        }
+
+        protected ResourcesManagementClient GetResourceManagementClient()
+        {
+            var options = Recording.InstrumentClientOptions(new ResourcesManagementClientOptions());
+            options.AddPolicy(CleanupPolicy, HttpPipelinePosition.PerCall);
+
+            return CreateClient<ResourcesManagementClient>(
+                TestEnvironment.SubscriptionId,
+                TestEnvironment.Credential,
+                options);
+        }
+
+        protected ValueTask<Response<T>> WaitForCompletionAsync<T>(Operation<T> operation)
+        {
+            if (Mode == RecordedTestMode.Playback)
+            {
+                return operation.WaitForCompletionAsync(TimeSpan.FromSeconds(0), default);
+            }
+            else
+            {
+                return operation.WaitForCompletionAsync();
+            }
+        }
+
+        protected async Task CleanupResourceGroupsAsync()
+        {
+            if (CleanupPolicy != null && Mode != RecordedTestMode.Playback)
+            {
+                var resourceGroupsClient = new ResourcesManagementClient(
+                    TestEnvironment.SubscriptionId,
+                    TestEnvironment.Credential,
+                    new ResourcesManagementClientOptions()).ResourceGroups;
+                foreach (var resourceGroup in CleanupPolicy.ResourceGroupsCreated)
+                {
+                    await resourceGroupsClient.StartDeleteAsync(resourceGroup);
+                }
+                CleanupPolicy.ResourceGroupsCreated.Clear();
+            }
+        }
```

Comment:
nit: remove the empty line. Can you open in VS 2019 and do a code format? Ctrl+E, Ctrl+D or Ctrl K, D if on different keyboard binding.